src/client/datascience/interactive-window/nativeInteractiveWindowCommandListener.ts

```diff
@@ -454,7 +466,7 @@ export class NativeInteractiveWindowCommandListener {
 
     private export(uri?: Uri) {
         const interactiveWindow = uri
-            ? this.interactiveWindowProvider.windows.find((window) => window.notebookUri!.toString() === uri.toString())
+            ? this.interactiveWindowProvider.windows.find((window) => window.notebookUri?.toString() === uri.toString())
```

Comment:
Maybe this 'find matching interactive window' could be on the intearctiveWindowProvider? And then everyone of these functions would use that code instead?

Reply:
Unfortunately we're still using the `IInteractiveWindowProvider` interface which is shared for webview, so I'd have to put that method on the old provider too (and it doesn't make sense there). I tracked this item in a list of todos for webview code cleanup https://github.com/microsoft/vscode-jupyter/issues/6488 feel free to add more there as well :)